Calibration/TkAlCaRecoProducers/plugins/TrackDistanceValueMapProducer.cc

```diff
@@ -120,7 +120,9 @@ void TrackDistanceValueMapProducer::produce(edm::StreamID streamID,
 
     // just copy the first nth
     std::vector<float> reduced_vdR2;
-    std::copy(v_dR2.begin(), v_dR2.begin() + nthClosestTrack_, std::back_inserter(reduced_vdR2));
+    if (!v_dR2.empty()) {
```

Comment:
Sorry I was bit sloppy when I wrote that it should be checked for `> 0`.  This probably ought to be something like
```
    std::copy(v_dR2.begin(), std::min(v_dR2.size(), v_dR2.begin() + nthClosestTrack_), std::back_inserter(reduced_vdR2));
```

Reply:
given `nthClosestTrack_` is configured as 1, I still don't quite understand how it is possible to have an event with one perfectly reconstructed muon and no other tracks in it...

Reply:
It'd have to be an event where the v_dR2 vector ends up empty, and it's probably relevant that the wf where this happens, 11634.24, has `--magField 0T`

Reply:
ugh, need more coffee...should be
```
    std::copy(v_dR2.begin(),  v_dR2.begin() + std::min(v_dR2.size(), nthClosestTrack_), std::back_inserter(reduced_vdR2));
```
and you'll probably have to change the type of `nthClosestTrack_` or static_cast or add a temporary so the types match.

Reply:
> It'd have to be an event where the v_dR2 vector ends up empty, 

that's what I don't understand how it can happen. It means there is a muon around which there is absolutely no track reconstructed. I'll have a look to this workflow to understand how exactly this happens. In the meanwhile I committed the changes you proposed.